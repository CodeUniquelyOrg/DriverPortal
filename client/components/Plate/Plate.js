import React, { Component } from 'react';
import PropTypes from 'prop-types';

import style from './style.css';

class Plate extends Component {
  static propTypes = {
    registration: PropTypes.string.isRequired,
    isYellow: PropTypes.bool,
  };

  static defaultProps = {
    isYellow: false,
  };

  render() {
    const {
      registration,
      isYellow,
      ...rest
    } = this.props;
    const styleClass = isYellow ? style.yellow : style.white;
    return (
      <div className={styleClass}>
        {registration}
      </div>
    );
  }
}

export default Plate;
