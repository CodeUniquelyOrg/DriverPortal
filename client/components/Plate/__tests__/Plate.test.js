import React from 'react';
import { shallow } from 'enzyme';

import Plate from 'components/Plate';

//
// Plate.js tests
//
describe('Plate', () => {
  //
  test('should render correctly', () => {
    const registration = {
    };
    const component = shallow(<Plate registration="XX00 XXX" />);
    expect(component).toMatchSnapshot();
  });

  //
});
