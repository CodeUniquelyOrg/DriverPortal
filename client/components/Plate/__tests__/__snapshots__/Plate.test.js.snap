// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Plate should render correctly 1`] = `
<div
  className="white"
>
  XX00 XXX
</div>
`;
